File: backend/app/schemas/menu.py
from pydantic import BaseModel, ConfigDict


class MenuCreate(BaseModel):
    name: str
    description: str | None = None


class MenuUpdate(BaseModel):
    name: str | None = None
    description: str | None = None


class MenuResponse(BaseModel):
    menu_id: int
    restaurant_id: int
    name: str
    description: str | None = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/types/menu.ts
export interface Menu {
    menu_id: number;
    restaurant_id: number;
    name: string;
    description: string | null;
}

export interface MenuCreate {
    name: string;
    description?: string | null;
}

export interface MenuUpdate {
    name?: string;
    description?: string | null;
}